hybrasyl/Board.cs

```diff
@@ -0,0 +1,319 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data.SqlTypes;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.Scripting.Interpreter;
+using Newtonsoft.Json;
+
+namespace Hybrasyl
+{
+     
+    public class MessageStoreLocked : Exception
+    {
+
+    }
+
+    [JsonObject(MemberSerialization.OptIn)]
+    public class MessageStore : IEnumerable<Message>
+    {
+        [JsonProperty] public string Name;
+        [JsonProperty] public string DisplayName;
+        [JsonProperty] public List<Message> Messages;
+        [JsonProperty] public string Guid;
+        [JsonProperty] public short CurrentId;
+        public int Id;
+
+        public bool Full => Messages.Count == short.MaxValue;
+        private int _lock; 
+        public bool IsSaving;
```

Comment:
public properties should implement a getter/setter. This assists with dependency injection. Really a non issue, but I would change this (example) to

`private bool _isSaving;

public bool IsSaving
{
     get
         {
              return _isSaving;
         }
     set
         {
            _isSaving = value;
         }
}
`